app/src/main/java/com/scouts/kitchenplaner/model/usecases/EditRecipe.kt

```diff
@@ -0,0 +1,223 @@
+     */
+    suspend fun setRecipePicture(recipe: Recipe, newURI: Uri) {
+        recipeRepository.setRecipeImage(recipe.id, newURI)
+    }
+
+    /**
+     * Set a recipe's description to a new one
+     *
+     * @param recipe The recipe to change
+     * @param newDescription The new description
+     */
+    suspend fun setRecipeDescription(recipe: Recipe, newDescription: String) {
+        recipeRepository.setRecipeDescription(recipe.id, newDescription)
+    }
+
+    /**
+     * Set a recipe's number of people it is calculated for to a new value
+     *
+     * @param recipe The recipe to change
+     * @param newNumberOfPeople The new number of people the recipe is calculated for
+     */
+    suspend fun setNumberOfPeople(recipe: Recipe, newNumberOfPeople: Int) {
+        recipeRepository.setNumberOfPeople(recipe.id, newNumberOfPeople)
+    }
+
+    /**
+     * Add a DietarySpeciality to a recipe
+     *
+     * @param recipe The recipe to which the speciality should be added
+     * @param speciality The speciality that should be added
+     */
+    suspend fun addDietarySpeciality(recipe: Recipe, speciality: DietarySpeciality) {
+        recipeRepository.insertDietarySpeciality(recipe.id, speciality.allergen, speciality.type)
+    }
+
+    /**
+     * Remove a DietarySpeciality from a recipe
+     *
+     * @param recipe The recipe from which to remove the speciality
+     * @param speciality The speciality that should be removed
+     */
+    suspend fun deleteDietarySpeciality(recipe: Recipe, speciality: DietarySpeciality) {
+        recipeRepository.deleteDietarySpeciality(recipe.id, speciality.allergen)
+    }
+
+    /**
+     * Add a new IngredientGroup to a recipe (Note that ingredient groups must not be empty)
+     *
+     * @param recipe The recipe to which to add the ingredient group
+     * @param group The IngredientGroup that should be added
+     */
+    suspend fun addIngredientGroup(recipe: Recipe, group: IngredientGroup) {
+        group.ingredients.forEach {
+            recipeRepository.insertIngredient(recipe.id, group.name, it)
+        }
+    }
+
+    /**
+     * Remove an IngredientGroup from a recipe
+     *
+     * @param recipe The recipe from which to remove the ingredient group
+     * @param group The ingredient group that should be removed
+     */
+    suspend fun deleteIngredientGroup(recipe: Recipe, group: IngredientGroup) {
+        recipeRepository.deleteIngredientGroup(recipe.id, group.name)
+    }
+
+    /**
+     * Add an ingredient to a recipe
+     *
+     * @param recipe The recipe to which to add the ingredient
+     * @param group The ingredient group to which to add the ingredient
+     * @param ingredient The ingredient that should be added
+     */
+    suspend fun addIngredient(recipe: Recipe, group: IngredientGroup, ingredient: Ingredient) {
+        recipeRepository.insertIngredient(recipe.id, group.name, ingredient)
+    }
+
+    /**
+     * Remove an ingredient from a recipe
```

Comment:
Note that, if the this ingredient is the last ingredient in an ingredient group, this group will also be deleted